### PR TITLE
Add pyproject.toml and other edits

### PR DESCRIPTION
**Pull request by celinedurniak**

### CODE REVIEW

**Review by franzlang — COMMENTED** [review]
the change to use `pyproject.toml` and the various style improvements (for pylint and flake8) look good to me. Unfortunately, I can't make an informed judgement on some of the functional changes in the `QENSmodels` scripts. I will also try to find the time to run the Jupyter notebooks, as they are near-impossible to review using only the git diff.

**Review by sanghamitra-mukhopadhyay — APPROVED** [review]
I am approving this on the basis of reviewing changes made in the code. Those are good additions accompanied with proper comments. I have not followed steps mentioned in doc/examples/  to check whether those are workable instructions.

File: QENSmodels/delta_lorentz.py [hunk 1]
```diff
@@ -74,16 +73,33 @@ def sqwDeltaLorentz(
...
+    A0 = np.asarray(A0)
+    hwhm = np.asarray(hwhm)
 
     # Create output array
     sqw = np.zeros((q.size, w.size))
 
     # Model
     if q.size > 1:
+        # if only a single float is given for A0, adapt to size of q
+        if A0.size == 1:
```
Comment (franzlang) [inline QENSmodels/delta_lorentz.py:85]: `float` and `list` have no attribute `size` and will lead to an error if `A0` is passed as one of those types. maybe include a type check for `A0`

File: QENSmodels/delta_lorentz.py [hunk 2]
```diff
@@ -74,16 +73,33 @@ def sqwDeltaLorentz(
...
+            A0 = A0 * np.ones(q.size)
+        # else
+        # check that enough values of A0 are given to match the size of q
+        else:
+            assert A0.shape == q.shape, \
```
Comment (franzlang) [inline QENSmodels/delta_lorentz.py:90]: will need to double-check that only object types with a `shape` attribute actually get to this part of the logic to be checked.

File: QENSmodels/delta_lorentz.py (hunk 1 above)
Comment (celinedurniak) [reply QENSmodels/delta_lorentz.py:85]: `size` and `shape` are used on `numpy.array`. For example, about this comment, `A0` is converted to a `numpy.array` a few lines above (line 76). 
I checked that this was also the case in all models.

File: QENSmodels/delta_lorentz.py (hunk 2 above)
Comment (celinedurniak) [reply QENSmodels/delta_lorentz.py:90]: `q` and `A0` are converted to `numpy.array`s at lines 75, 76, respectively.

**Review by franzlang — APPROVED** [review]
thanks for the clarification and sorry for missing the obvious conversion to an np.array. I am happy for it to be merged by someone with write access.

**Comment by celinedurniak** [comment]
Thanks @franzlang and @sanghamitra-mukhopadhyay for the reviews.

**Pull request merged**